packages/progressive-profiling-nodejs/src/implementation.ts

```diff
@@ -0,0 +1,314 @@
+import {
+  RegisterSections,
+  FormSection,
+  SuperTokensPluginProfileProgressiveProfilingNormalisedConfig,
+  UserMetadataConfig,
+} from "./types";
+import { logDebugMessage } from "./logger";
+import { FormField, FormFieldValue, ProfileFormData } from "@supertokens-plugins/progressive-profiling-shared";
+import { SessionContainerInterface } from "supertokens-node/recipe/session/types";
+import { BooleanClaim } from "supertokens-node/recipe/session/claims";
+import { PLUGIN_ID, METADATA_KEY, METADATA_PROFILE_KEY } from "./constants";
+import { pluginUserMetadata } from "@shared/nodejs";
+import { BasePluginImplementation, groupBy, indexBy, mapBy } from "@shared/js";
+
+export class Implementation extends BasePluginImplementation<SuperTokensPluginProfileProgressiveProfilingNormalisedConfig> {
+  protected existingSections: (FormSection & { registratorId: string })[] = [];
+  protected existingRegistratorHandlers: Record<string, Pick<Parameters<RegisterSections>[0], "set" | "get">> = {};
+  protected metadata = pluginUserMetadata<{ profileConfig?: UserMetadataConfig }>(METADATA_KEY);
+
+  static ProgressiveProfilingCompletedClaim: BooleanClaim;
+
+  constructor(pluginConfig: SuperTokensPluginProfileProgressiveProfilingNormalisedConfig) {
+    super(pluginConfig);
+
+    Implementation.ProgressiveProfilingCompletedClaim = new BooleanClaim({
+      key: `${PLUGIN_ID}-completed`,
+      fetchValue: async (userId, recipeUserId, tenantId, currentPayload, userContext) => {
+        const userMetadata = await this.metadata.get(userId);
+        return this.areAllSectionsCompleted(
+          // can't pass session here because it's not available in the params or a way of getting it
+          undefined as unknown as SessionContainerInterface,
+          userMetadata?.profileConfig,
+          userContext,
+        );
+      },
+    });
+  }
+
+  // todo make sure the implementation is the same as in the profile plugin (when it will be implement in the new repo - maybe part of a shared library or exported from the plugin itself ?)
+  getDefaultRegistrator = function (
```

Comment:
This feels a bit weird. What do people achieve by overriding this?
Would it make sense to split this into `defaultGetSection` and `defaultSetSection` or something similar?